### PR TITLE
changes to remove unwanted calculations

### PR DESCRIPTION
**Pull request by anupsv**
Fixing kzg calculations and removing unwanted things and aligning the remaining of the data with the data formatting.

### CODE REVIEW

File: arbitrator/prover/src/kzgbn254.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ pub fn prove_kzg_preimage_bn254(
     let blob_commitment = kzg.commit(&blob_polynomial_evaluation_form)?;
 
     let mut commitment_bytes = Vec::new();
-    blob_commitment.serialize_uncompressed(&mut commitment_bytes)?;
+    blob_commitment.serialize_uncompressed(&mut commitment_bytes)?; // why uncompressed ?
```
Comment (bxue-l2) [inline arbitrator/prover/src/kzgbn254.rs]: true, even kzg.rs used compressed I think, https://github.com/Layr-Labs/nitro/blob/eigenda-v3.0.3/arbitrator/prover/src/kzg.rs#L75C55-L75C65

File: arbitrator/prover/src/kzgbn254.rs [hunk 2]
```diff
@@ -69,39 +69,29 @@ pub fn prove_kzg_preimage_bn254(
...
-    assert!(length_usize / 32 == preimage_polynomial.len() as u64);
+    assert!(length_usize / 32 == blob_polynomial_evaluation_form.len() as u64);
 
     // address proving past end edge case later
     let proving_past_end = offset as u64 >= length_usize;
     if proving_past_end {
         // Proving any offset proves the length which is all we need here,
         // because we're past the end of the preimage.
-        proving_offset = 0;
+        proving_offset = 0; // why 0? shouldn't this fail immediately ?
```
Comment (bxue-l2) [inline arbitrator/prover/src/kzgbn254.rs]: The kzg.rs also set it as 0, I guess it is a default setting. As long as all software are implemented with the same spec, we are fine?

File: arbitrator/prover/src/kzgbn254.rs [hunk 3]
```diff
@@ -124,7 +115,17 @@ pub fn prove_kzg_preimage_bn254(
     let g2_tau_minus_g2_z = (g2_tau - z_g2).into_affine();
 
     let kzg_proof = kzg
-        .compute_kzg_proof_with_roots_of_unity(&preimage_polynomial, proving_offset as u64 / 32)?;
+        .compute_kzg_proof_with_roots_of_unity(&blob_polynomial_evaluation_form, proving_offset as u64 / 32)?;
+    
+    // This should cause failure when proving past offset.
+    if !proving_past_end {
```
Comment (bxue-l2) [inline arbitrator/prover/src/kzgbn254.rs:123]: why removing it? this can still happen, when blob has size of 0

Comment (anupsv) [reply arbitrator/prover/src/kzgbn254.rs:123]: Ah not removing it, just commented it for testing locally.

File: arbitrator/prover/src/kzgbn254.rs (hunk 2 above)
Comment (anupsv) [reply arbitrator/prover/src/kzgbn254.rs]: Discussed the reasoning on slack. This is handled on the contract side and done as "0" here to let the code progress and complete.

File: arbitrator/prover/src/kzgbn254.rs [hunk 4]
```diff
@@ -69,39 +68,29 @@ pub fn prove_kzg_preimage_bn254(
...
-    assert!(length_usize / 32 == preimage_polynomial.len() as u64);
+    assert!(length_usize / 32 == blob_polynomial_evaluation_form.len() as u64);
 
     // address proving past end edge case later
     let proving_past_end = offset as u64 >= length_usize;
     if proving_past_end {
         // Proving any offset proves the length which is all we need here,
         // because we're past the end of the preimage.
-        proving_offset = 0;
+        proving_offset = 0; // why 0? shouldn't this fail immediately ?
```
Comment (bxue-l2) [inline arbitrator/prover/src/kzgbn254.rs]: maybe remove the comment why 0?

**Review by bxue-l2 — APPROVED** [review]

File: arbitrator/prover/src/kzgbn254.rs (hunk 4 above)
Comment (anupsv) [reply arbitrator/prover/src/kzgbn254.rs]: sure.

**Pull request merged**